universal/include/userver/formats/parse/try_parse.hpp

```diff
@@ -28,25 +28,33 @@ inline bool Is(Value&& value) {
   }
 }
 
-inline constexpr utils::TypeList<bool, int, std::int64_t, std::uint64_t, double, std::string> kBaseTypes;
+inline constexpr utils::impl::TypeList<bool, std::int64_t, std::uint64_t, double, std::string> kBaseTypes;
 
 
 } // namespace impl
 
 
 template <typename T, typename Value>
-inline std::enable_if_t<utils::AnyOf(utils::IsSameCarried<T>(), impl::kBaseTypes), std::optional<T>>
+constexpr inline std::enable_if_t<utils::impl::AnyOf(utils::impl::IsConvertableCarried<T>(), impl::kBaseTypes), std::optional<T>>
```

Comment:
If the type is merely convertible to `bool`, `double` or `string`, though, we don't want to allow parsing it using the conversion, to be consistent with `Parse` and `Serialize`